### DISCUSSION TITLE
VIPS Image is skewed ...

### DISCUSSION THREAD

**Discussion by ademmler** [libvips/libvips#2026] [Q&A]
Hello VIPs professionals.

I am trying to load an image with vips and display this in Qt. After loading the image it is skewed.

Attached a minimal code sample and a test image. 
Also two screenshots from using vips or qt internal function to load image.

Any hint and help is welcome.
thx in advance Alexander

[VipsMinimal.zip](https://github.com/libvips/libvips/files/4993952/VipsMinimal.zip)

**Comment by jcupitt** [comment]
Hello @ademmler, have you seen the official display demo? It's gtk rather than qt, but it should be simple to adapt.

https://github.com/libvips/vipsdisp-tiny

**Comment by jcupitt** [comment]
In your code, you are using `QImage(pointer, width, height, format)`. Perhaps this has assumptions about scanline padding? I would try the constructor that lets you set bytesperline as well. Use `VIPS_IMAGE_SIZEOF_LINE(image.get_image())` to get the line size.

**Comment by jcupitt** [comment]
There's a more fleshed-out image viewer here:

https://github.com/jcupitt/vipsdisp

Though it's not quite finished. It looks like this:

![x](https://user-images.githubusercontent.com/580843/88789173-4fa50f00-d18e-11ea-9807-eafa5f289fcb.png)

**Comment by ademmler** [comment]
> Hello @ademmler, have you seen the official display demo? It's gtk rather than qt, but it should be simple to adapt.
> 
> https://github.com/libvips/vipsdisp-tiny

Yes - I downloaded this and got it compiled. But for now I could not see the key to solve it.

**Comment by ademmler** [comment]
Hello @jcupitt,

Your hint did the trick ...

/*
* Hint from vips guy for fixinf skewing:
* In your code, you are using QImage(pointer, width, height, format).
* Perhaps this has assumptions about scanline padding?
* I would try the constructor that lets you set bytesperline as well.
* Use VIPS_IMAGE_SIZEOF_LINE(image.get_image()) to get the line size.
*/

The code:
```
vipsObject = VImage::new_from_file (imagePath.toStdString().c_str(), VImage::option ()->set ("access", VIPS_ACCESS_SEQUENTIAL));

   imageObject = new QImage();
   *imageObject = QImage((uchar*)vipsObject.data(), vipsObject.width(), vipsObject.height(), VIPS_IMAGE_SIZEOF_LINE(vipsObject.get_image()), QImage::Format_RGB888);
   image = QPixmap::fromImage(*imageObject);
```
